README.md

```diff
@@ -10,62 +10,86 @@ Please find project details at [Wiki](https://github.com/Samsung/TizenRT/wiki) e
+However, if your development systems are not eligible for running the Docker container, all libraries and toolchains should be manually installed.  
+Please refer to [Manual Setup Build Environment](docs/HowToSetEnv.md).
 
-```bash
-tar xvjf gcc-arm-none-eabi-6-2017-q1-update-[OS Type].tar.bz2
-export PATH=<Your Toolchain PATH>:$PATH
-```
-Be aware that recommanded toolchain is fully working on 64bits machine.
+For more infomation of libraries in the TizenRT Docker Image, see https://hub.Docker.com/r/tizenrt/tizenrt/.
+
+### 1. Install Docker
 
-NOTE: See [Additional Requirements](docs/InstallAdReq.md) for gRPC, ROMFS, OCF and IoT.js.
+To install OS specific Docker engines, see https://docs.docker.com/install/linux/docker-ce/ubuntu/.  
+If you already have a Docker engine, please skip this step.
 
-### Getting the sources
+### 2. Getting TizenRT source code
 
 ```bash
 git clone https://github.com/Samsung/TizenRT.git
 cd TizenRT
 TIZENRT_BASEDIR="$PWD"
 ```
+**Note**: To contribute in this community, fork this project first into own github repository and do git cloning from your repository. 
+> git clone https://github.com/your_github_id/TizenRT.git
+
+
+### 3. Build
 
-### How to Build
+#### 3.1 Configuration (*$TIZENRT_BASEDIR/os/tools*)
 
-Configure the build from *$TIZENRT_BASEDIR/os/tools* directory
 ```bash
 cd os/tools
 ./configure.sh <board>/<configuration_set>
 ```
-The configuration file is named *defconfig*,  
-and resides under the relative path \<board\>/\<configuration_set\> rooted at *build/configs*.  
-To check the different \<board\>/\<configuration_set\> combinations supported, type below:
+
+This command retrieves the specific pre-configured file named defconfig according to
+ \<board\>/\<configuration_set\>.  
+You can see collection of all configuration files at
+ *$TIZENRT_BASEDIR/build/configs*.  
+To check all pre-defined configurations, type as follows:
+
 ```bash
 ./configure.sh --help
 ```
 
-After configuring above, configuration can be modified through *make menuconfig* from *$TIZENRT_BASEDIR/os*.
+#### 3.2 Additional Configuration (*$TIZENRT_BASEDIR/os*)
+
+After the configuration is done, further modification can be done with menuconfig from
+ *$TIZENRT_BASEDIR/os*.  
+This command requires ```sudo``` for root permission.  
+To run Docker without ```sudo```, refer to
+ https://docs.docker.com/install/linux/linux-postinstall/.
+
 ```bash
 cd ..
-make menuconfig
+./dbuild.sh menuconfig
+```
+
+#### 3.3 Compile (*$TIZENRT_BASEDIR/os*)
+
+```bash
+./dbuild.sh
 ```
 
-Refer [kconfig-frontend installation](docs/HowtoInstallKconfigFrontend.md) to use *menuconfig*.
+Built binaries are located in *$TIZENRT_BASEDIR/build/output/bin*.
+
+#### 3.4 Clean (*$TIZENRT_BASEDIR/os*)
 
-Finally, initiate build by make from *$TIZENRT_BASEDIR/os*.
 ```bash
-make
+./dbuild.sh clean
 ```
+This command removes built files including objects, libraries, .depend, Make.dep, etc.  
```

Comment:
We know these two commands what they are. But beginner is difficult to know when it should be called.
How about below?

There are two clean commands, *clean* and *distclean*.
```bash
./dbuild.sh clean
```
This command removes built files including objects, libraries, .depend, Make.dep, etc. 
After modifying configuration with menuconfig or to build all of module again, this command is required.

```bash
./dbuild.sh distclean
```
This command executes *clean* command, and additionally it removes configured files including .config, Make.defs and all of symbolic linked files.
Before changing configuration with ```./configure.sh``` command, deletion of previous configuration files must be done using the distclean command.